src/storage/gpu_swap_prefetch.cc

```diff
@@ -74,12 +81,13 @@ void Prefetch::Prefetching(int device) {
 void Prefetch::HistoryBasedPrefetch(int device) {
   //pthread_rwlock_rdlock(&swap_lock_);
   //bool has_begun = false;
-  while(lookahead_pos_[device]+1 < history_->ordered_history[device].size() 
-      && lookahead_pos_[device] - history_->record_idx[device] <= steps_ahead_) {
+  while(lookahead_pos_[device]+1 < history_->ordered_history[device].size() &&
+      lookahead_pos_[device]-(int)history_->record_idx[device] <= steps_ahead_) {
```

Comment:
If this is to avoid overflow, maybe it is better to change it to `lookahead_pos_[device] <= steps_ahead_ + history_->record_idx[device]` so that no type conversion is necessary